### PR TITLE
Add mergesort

### PR DESCRIPTION
**Pull request by mbalamat**
check my mergesort

### CODE REVIEW

File: mergesort/mergesort.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+def merge(A,B):
+    i=0
+    j=0
+    TEMP=[]
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use uppercase variable names only for constants.

File: mergesort/mergesort.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+def merge(A,B):
+    i=0
+    j=0
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use spaces before and after '='.

File: mergesort/mergesort.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    TEMP=[]
+    for k in range(len(A+B)):
+        if (i<len(A) and j<len(B)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: The parentheses here are redundant.

File: mergesort/mergesort.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+def merge(A,B):
+    i=0
+    j=0
+    TEMP=[]
+    for k in range(len(A+B)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use spaces before and after '+'.

File: mergesort/mergesort.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        if (i<len(A) and j<len(B)):
+            if A[i]<B[j]:
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use spaces before and after '<'.

File: mergesort/mergesort.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                TEMP.append(A[i])
+                i += 1
+            else:
+                TEMP.append(B[j])
+                j += 1
+        elif i>=len(A) and j<len(B):
+            while (j<len(B)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: Why not just TEMP += B[j:]?

File: mergesort/mergesort.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                TEMP.append(B[j])
+                j += 1
+        else:
+            while (i<len(A)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: Same as line 14.

File: mergesort/mergesort.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                TEMP.append(A[i])
+                i += 1
+    return TEMP
+
+def mergeSort(A):
+    n = len(A)
+    if n == 1:
+        return A #it is sorted
+    mid = n//2
+    leftHalf=A[:mid]
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use spaces before and after '='.

File: mergesort/mergesort.py (hunk 1 above)
Comment (gtklocker) [inline mergesort/mergesort.py]: Use a more meaningful variable name.

Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Use spaces before and after '='.

File: mergesort/mergesort.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+            while (i<len(A)):
+                TEMP.append(A[i])
+                i += 1
+    return TEMP
+
+def mergeSort(A):
+    n = len(A)
+    if n == 1:
+        return A #it is sorted
+    mid = n//2
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Spaces.

File: mergesort/mergesort.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
+def merge(a,b):
+    i = 0
+    j = 0
+    temp = []
+    for k in range(len(a+b)):
+        if i < len(a) and j < len(b):
+            if a[i]<b[j]:
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Spaces.

File: mergesort/mergesort.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            while (i < len(a)):
+                temp.append(a[i])
+                i += 1
+    return temp
+
+def mergeSort(a):
+    n = len(a)
+    if n == 1:
+        return a #it is sorted
+    mid = n//2
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Spaces.

File: mergesort/mergesort.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
+def merge(a,b):
+    i = 0
+    j = 0
+    temp = []
+    for k in range(len(a+b)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Spaces.

File: mergesort/mergesort.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        if i < len(a) and j < len(b):
+            if a[i]<b[j]:
+                temp.append(a[i])
+                i += 1
+            else:
+                temp.append(b[j])
+                j += 1
+        elif i >= len(a) and j < len(b):
+            while (j < len(b)):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: Why did you remove temp += b[j:]? It's good enough, you just have to break the for loop afterwards.

File: mergesort/mergesort.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
+def merge(a,b):
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Spaces.

File: mergesort/mergesort.py (hunk 13 above)
Comment (mbalamat) [inline mergesort/mergesort.py]: because I like it !!!!

Comment (gtklocker) [inline mergesort/mergesort.py]: Change it back if you want it merged.

File: mergesort/mergesort.py [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+                temp.append(a[i])
+                i += 1
+            else:
+                temp.append(b[j])
+                j += 1
+        elif i >= len(a) and j < len(b):
+            temp += b[j:]
+            break
+        else:
+           temp += a[i:]
```
Comment (gtklocker) [inline mergesort/mergesort.py]: nit: Should be 4 spaces indented.

**Comment by gtklocker** [comment]
Your commit history is all dirty. Can you please [squash your commits](http://davidwalsh.name/squash-commits-git) into a single one?

**Comment by gtklocker** [comment]
Looking good, thanks! :+1:
